config_parser_test.cc

```diff
@@ -9,3 +9,61 @@ TEST(NginxConfigParserTest, SimpleConfig) {
 
   EXPECT_TRUE(success);
 }
+
+//test nested directives
+TEST(NginxConfigParserTest, NestedConfig) {
+  NginxConfigParser parser;
+  NginxConfig out_config;
+  bool success = parser.Parse("config_with_nested", &out_config);
+
+  EXPECT_EQ(out_config.statements_.size(), 1);
+  EXPECT_TRUE(success); //this was fixed to allow nested Configs that end like }}
+}
+
+TEST(NginxConfigParserTest, LargeConfig) {
+  NginxConfigParser parser;
+  NginxConfig out_config;
+
+  EXPECT_TRUE(parser.Parse("large_config", &out_config));
+}
+
+//Test fixture for string parsing
+class NginxStringConfigTest : public ::testing::Test {
+protected:
+    bool parseString(const std::string config_string) {
+        std::stringstream config_stream(config_string);
+        return parser.Parse(&config_stream, &out_config);
+    }
+
+    NginxConfigParser parser;
+    NginxConfig out_config;
+};
+
+//test invalid inputs
+TEST_F(NginxStringConfigTest, InvalidConfig) {
+    EXPECT_FALSE(parseString("foo bar"));
+    EXPECT_FALSE(parseString("foo {\n}"));
+    EXPECT_FALSE(parseString("foo {\n\tx;\n};"));
```

Comment:
I'll add comments to each of these to explain as well, but I think its debatable whether foo {} is an error; I don't see any reason a programmer would want to leave it empty unless they forgot to put something in there. We can fix this later if necessary anyway, I'll just leave it as is for now.